src/server/api/routers/media-request.ts

```diff
@@ -24,10 +24,17 @@ export const mediaRequestsRouter = createTRPCRouter({
         checkIntegrationsType(app.integration, ['overseerr', 'jellyseerr'])
       );
 
-      const promises = apps.map((app): Promise<MediaRequest[]> => {
+      const promises = apps.map(async (app): Promise<MediaRequest[]> => {
         const apiKey =
           app.integration?.properties.find((prop) => prop.field === 'apiKey')?.value ?? '';
         const headers: HeadersInit = { 'X-Api-Key': apiKey };
+
+        // Get Version
+        const versionResponse = await fetch(`${app.url}/api/v1/status`, { headers });
+        const versionBody = await versionResponse.json();
+        const version = versionBody.version || '0.0.0';
```

Comment:
- ``0.0.0`` probably isn't a good fallback value
- Is a fallback value a good idea? Shouldn't we throw if we can't find the version? And from a different perspective, is there even a possibility for this to be undefined?

Reply:
How about modifying it like this？
```
        const versionResponse = await fetch(`${app.url}/api/v1/status`, { headers });
        if (!versionResponse.ok) {
          throw new Error(`Failed to get version: ${versionResponse.statusText}`);
        }
        const versionBody = await versionResponse.json();
        const version = versionBody.version;
        if (typeof version === 'undefined') {
          throw new Error('Version undefined');
        }
```